File: backend/app/schemas/store.py
from datetime import date

from pydantic import BaseModel, Field


class StoreCreateRequest(BaseModel):
    business_name: str = Field(min_length=1, max_length=100)
    industry_name: str = Field(min_length=1, max_length=100)
    business_address: str = Field(min_length=1, max_length=255)
    open_date: date


class StoreResponse(BaseModel):
    store_id: int
    business_name: str
    industry_name: str
    business_address: str
    open_date: date

    model_config = {
        "from_attributes": True
    }

class StoreDetailResponse(BaseModel):
    business_name: str
    industry_name: str
    business_address: str
    open_date: date
    business_period: str

File: frontend/src/types/store.ts
/** POST /api/v1/stores · GET /api/v1/stores/me 요청·응답. */

export interface StoreCreateRequest {
  business_name: string
  industry_name: string
  business_address: string
  /** 개업일 YYYY-MM-DD */
  open_date: string
}

export interface StoreResponse {
  store_id: number
  business_name: string
  industry_name: string
  business_address: string
  open_date: string
}

export interface StoreDetailResponse {
  business_name: string
  industry_name: string
  business_address: string
  open_date: string
  /** 서버가 계산한 사업기간. 예: "3년 2개월" */
  business_period: string
}
